internal/receiver/scriptedinputsreceiver/operator.go

```diff
@@ -0,0 +1,167 @@
+// Copyright Splunk, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scriptedinputsreceiver
+
+import (
+	"bufio"
+	"context"
+	"io"
+	"sync"
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/helper"
+	"go.uber.org/zap"
+)
+
+func init() {
+	operator.Register(operatorType, func() operator.Builder { return createDefaultConfig() })
+}
+
+const (
+	// DefaultMaxLogSize is the max buffer sized used
+	// if MaxLogSize is not set
+	defaultMaxLogSize = 1024 * 1024
+	operatorType      = "procpipe_input"
+)
+
+// stdoutOperator is an operator that reads input from stdout
+type stdoutOperator struct {
+	cfg           *Config
+	logger        *zap.SugaredLogger
+	cancelAll     context.CancelFunc
+	splitFunc     bufio.SplitFunc
+	decoder       *helper.Decoder
+	scriptContent string
+	helper.InputOperator
+	wg sync.WaitGroup
+}
+
+// Start will start generating log entries.
+func (i *stdoutOperator) Start(_ operator.Persister) error {
+
+	ctx, cancelAll := context.WithCancel(context.Background())
+	i.cancelAll = cancelAll
+
+	ticker := time.NewTicker(i.cfg.interval)
+
+	go func() {
+		for {
+			_, cancelCycle := context.WithCancel(ctx)
```

Comment:
not using this context is still an issue, since the `cancelCycle()` would only influence it.

Reply:
What do you propose as remediation?

Reply:
I'd assume you want to pass the copy to `i.beginCycle`: https://pkg.go.dev/context#WithCancel